backend/sozisel/lib/sozisel/model/event_results.ex

```diff
@@ -47,4 +50,116 @@ defmodule Sozisel.Model.EventResults do
   def change_event_result(%EventResult{} = event_result, attrs \\ %{}) do
     EventResult.create_changeset(event_result, attrs)
   end
+
+  def get_all_event_results(%LaunchedEvent{} = launched_event) do
+    launched_event
+    |> Repo.preload(:event_results)
+    |> then(& &1.event_results)
+  end
+
+  def quiz_summary(%LaunchedEvent{} = event_launched) do
+    quiz_results = get_all_event_results(event_launched)
+
+    total_points =
+      quiz_results
+      |> Enum.flat_map(& &1.result_data.participant_answers)
+      |> Enum.map(& &1.points)
+      |> Enum.sum()
+
+    total_quiz_answer_time =
+      quiz_results |> Enum.map(& &1.result_data.quiz_answer_time) |> Enum.sum()
+
+    participant_answers =
+      quiz_results
+      |> Enum.flat_map(& &1.result_data.participant_answers)
+
+    %{
+      number_of_participants: length(quiz_results),
+      average_points: total_points / length(participant_answers),
+      average_quiz_answer_time: total_quiz_answer_time / length(quiz_results)
+    }
+  end
+
+  def quiz_participants_summary(%LaunchedEvent{} = event_launched) do
+    quiz_results = get_all_event_results(event_launched)
+
+    quiz_participant_summary =
+      quiz_results
+      |> Enum.map(fn participant_result ->
+        result_data =
+          participant_result
+          |> Utils.from_deep_struct()
+          |> Map.get(:result_data)
```

Comment:
maybe try first to do the `Map.get`? This function from_deep_struct otherwise has a lot of work to do